### PR TITLE
Ansible_Job api test coverage [e2e]

### PR DESCRIPTION
**Pull request by adarshdubey-star**
**Test results:**

```
pytest tests/foreman/api/test_ansible.py -k test_positive_ansible_job_on_host
============================================================== test session starts ===============================================================
platform linux -- Python 3.8.13, pytest-7.2.0, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/addubey/work/robottelo, configfile: pyproject.toml
plugins: forked-1.4.0, services-2.2.1, cov-3.0.0, xdist-3.1.0, ibutsu-2.2.4, reportportal-5.1.3, mock-3.10.0
collected 4 items / 1 deselected / 3 selected                                                                                                    

tests/foreman/api/test_ansible.py ...                                                                                                      [100%]
============================================ 3 passed, 1 deselected, 5 warnings in 1487.51s (0:24:47) ============================================

```
Signed-off-by: Adarsh Dubey <addubey@redhat.com>

### CODE REVIEW

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k test_positive_ansible_job_on_host
nailgun: 841

**Review by synkd — COMMENTED** [review]
Looks like your PRT run failed because https://github.com/SatelliteQE/nailgun/pull/841 does not contain the changes that you made in https://github.com/SatelliteQE/nailgun/pull/866. So, while I trust your local results and the test in this PR looks good to me, the PRT failure does point to the fact that this test depends on both of the above nailgun PRs. Should we hold off on merging  this until nailgun 841 is merged?

**Comment by adarshdubey-star** [comment]
> Looks like your PRT run failed because [SatelliteQE/nailgun#841](https://github.com/SatelliteQE/nailgun/pull/841) does not contain the changes that you made in [SatelliteQE/nailgun#866](https://github.com/SatelliteQE/nailgun/pull/866). So, while I trust your local results and the test in this PR looks good to me, the PRT failure does point to the fact that this test depends on both of the above nailgun PRs. Should we hold off on merging this until nailgun 841 is merged?

Yes @synkd 841 should go in first as this test case utilizes those endpoints.

File: tests/foreman/api/test_ansible.py
Comment (Griffin-Sullivan) [inline tests/foreman/api/test_ansible.py:100]: This is minor but a more descriptive variable name here would be better like `proxy_id`

**Review by Griffin-Sullivan — APPROVED** [review]
Awesome coverage to get in for stream testing! We could even consider that this is testing RH Cloud - Insights as a sub-component if we ever want to start adding metadata for sub-component.

**Review by synkd — APPROVED** [review]

**Review by pondrejk — APPROVED** [review]

File: tests/foreman/api/test_ansible.py
Comment (Gauravtalreja1) [inline tests/foreman/api/test_ansible.py:109]: can you try nailgun_host here?
```suggestion
    assert rhel_contenthost.nailgun_host.name == rhel_contenthost.hostname
```

File: tests/foreman/api/test_ansible.py
Comment (Gauravtalreja1) [inline tests/foreman/api/test_ansible.py:125]: Instead of importing it from API utils, would suggest using it from sat obj
```suggestion
    target_sat.wait_for_tasks(
        f'resource_type = JobInvocation and resource_id = {job["id"]}', poll_timeout=1000
    )
```

File: tests/foreman/api/test_ansible.py
Comment (Gauravtalreja1) [reply tests/foreman/api/test_ansible.py:100]: @adarshdubey-star Replacing internal_capsule with nailgun_smart_proxy property in Capsule hosts class, if this PR https://github.com/SatelliteQE/robottelo/pull/10458 gets in first, then we'll need to update this.

File: tests/foreman/api/test_ansible.py
Comment (Gauravtalreja1) [inline tests/foreman/api/test_ansible.py:90]: CaseAutomation already defined at module level, this can be removed

File: tests/foreman/api/test_ansible.py
Comment (Gauravtalreja1) [inline tests/foreman/api/test_ansible.py:78]:
```suggestion
```

File: tests/foreman/api/test_ansible.py
Comment (Gauravtalreja1) [inline tests/foreman/api/test_ansible.py:70]: can we include fips enabled hosts as well here?

**Review by ColeHiggins2 — APPROVED** [review]
Ack pending @Gauravtalreja1 comments

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k test_positive_ansible_job_on_host

File: tests/foreman/api/test_ansible.py
Comment (Gauravtalreja1) [inline tests/foreman/api/test_ansible.py:106]:
```suggestion
    assert target_host.name == rhel_contenthost.hostname
```

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k test_positive_ansible_job_on_host
nailgun: 881

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k test_positive_ansible_job_on_host
nailgun: 881

File: tests/foreman/api/test_ansible.py
Comment (adarshdubey-star) [reply tests/foreman/api/test_ansible.py:89]: I want to keep this one in the future if I have nonautomated scenarios It's beneficial to have a count for coverage.

File: tests/foreman/api/test_ansible.py
Comment (adarshdubey-star) [reply tests/foreman/api/test_ansible.py]: Addressed

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k test_positive_ansible_job_on_host
nailgun: 881

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k test_positive_ansible_job_on_host
nailgun: 881

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**
